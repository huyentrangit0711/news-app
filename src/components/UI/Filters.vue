<template>
	<div class="filter-container flex">
		<div
			class="mr-4 mb-4"
			v-for="(option, index) in avaiableOption"
			:key="index"
		>
			<input
				:id="index"
				type="radio"
				name="radio"
				class="hidden"
				:value="option"
				v-model="filterOptions"
			/>
			<label :for="index" class="flex items-center cursor-pointer text-xl">
				<span
					class="bg-white-600 bg-opacity-100 rounded-lg text-black p-2 border-black border"
				>
					{{ option }}
				</span>
			</label>
		</div>
	</div>
</template>

<script>
export default {
	name: 'Filters',
	data() {
		return {
			avaiableOption: ['bitcoin', 'apple', 'earthquake', 'animal'],
			filterOptions: null,
		};
	},
	components: {},
};
</script>

<style>
input[type='radio']:checked + label span {
	@apply bg-green-600 text-white;
}
input[type='radio'] + label span:hover,
input[type='radio'] + label:hover span {
	transform: scale(1.2);
}
</style>
