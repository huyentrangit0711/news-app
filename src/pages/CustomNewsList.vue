<template>
	<div class="filter-news-tab">
		<h1>News List With Filter Tab</h1>
		<filters></filters>
		<div class="grid grid-cols-3 gap-4">
			<new-item></new-item><new-item></new-item>
			<new-item></new-item>
			<new-item></new-item>
		</div>
	</div>
</template>

<script>
import Filters from '../components/UI/Filters.vue';
import NewItem from '../components/news/NewItem.vue';
export default {
	name: 'CustomNewListPage',
	components: {
		Filters,
		NewItem,
	},
};

// https://newsapi.org/v2/everything?q=  /// second tab - query with button filter on tab
// `animal`&sortBy=popularity&apiKey=
</script>
