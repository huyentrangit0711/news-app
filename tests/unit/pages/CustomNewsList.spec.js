import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import CustomNewsList from '@/pages/CustomNewsList';
import Filters from '@/components/UI/Filters';
const localVue = createLocalVue();

localVue.use(Vuex);
describe('Custom News List Page Component', () => {
	// render the component
	let wrapper;
	let actions;
	let store;
	beforeEach(() => {
		actions = {
			getNewsByQuery: jest.fn(),
		};
		let news = {
			namespaced: true,
			state: {},
			actions,
		};
		store = new Vuex.Store({
			modules: {
				news,
			},
		});
	});
	afterEach(() => {
		jest.clearAllMocks();
	});
	it('Page Custom News List should render correctly', () => {
		wrapper = shallowMount(CustomNewsList, {
			stubs: ['router-link', 'router-view'],
		});
		expect(wrapper).toMatchSnapshot();
	});
	it('dispatches "getNewsByQuery" when filter change ', () => {
		const wrapper = shallowMount(CustomNewsList, {
			store,
			localVue,
			stubs: ['router-link', 'router-view'],
		});
		const FiltersComponent = wrapper.findComponent(Filters);
		FiltersComponent.vm.$emit('filterChange', 'animal');
		expect(actions.getNewsByQuery).toHaveBeenCalled();
	});
});
