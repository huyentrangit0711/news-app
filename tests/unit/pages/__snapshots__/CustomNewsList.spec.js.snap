// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom News List Page Component Page Custom News List should render correctly 1`] = `
<div class="filter-news-tab">
  <h1>Top Random News</h1>
  <filters-stub></filters-stub>
  <router-view-stub></router-view-stub>
</div>
`;
